Zend/zend_alloc.c

```diff
@@ -1047,27 +1178,31 @@ static void *zend_mm_alloc_pages(zend_mm_heap *heap, uint32_t pages_count ZEND_F
 get_chunk:
 			if (heap->cached_chunks) {
 				heap->cached_chunks_count--;
+				ZEND_MM_POISON_CHUNK_HDR(chunk, heap);
 				chunk = heap->cached_chunks;
+				ZEND_MM_UNPOISON_CHUNK_HDR(chunk);
 				heap->cached_chunks = chunk->next;
 			} else {
 #if ZEND_MM_LIMIT
 				if (UNEXPECTED(ZEND_MM_CHUNK_SIZE > heap->limit - heap->real_size)) {
-					if (zend_mm_gc(heap)) {
+					if (_zend_mm_gc(heap)) {
 						goto get_chunk;
 					} else if (heap->overflow == 0) {
 #if ZEND_DEBUG
 						zend_mm_safe_error(heap, "Allowed memory size of %zu bytes exhausted at %s:%d (tried to allocate %zu bytes)", heap->limit, __zend_filename, __zend_lineno, size);
 #else
 						zend_mm_safe_error(heap, "Allowed memory size of %zu bytes exhausted (tried to allocate %zu bytes)", heap->limit, ZEND_MM_PAGE_SIZE * pages_count);
 #endif
+						ZEND_MM_POISON_CHUNK_HDR(chunk, heap);
 						return NULL;
 					}
 				}
 #endif
+				ZEND_MM_POISON_CHUNK_HDR(chunk, heap);
 				chunk = (zend_mm_chunk*)zend_mm_chunk_alloc(heap, ZEND_MM_CHUNK_SIZE, ZEND_MM_CHUNK_SIZE);
 				if (UNEXPECTED(chunk == NULL)) {
 					/* insufficient memory */
-					if (zend_mm_gc(heap) &&
+					if (_zend_mm_gc(heap) &&
```

Comment:
I know that PHP has quite a few identifiers starting with an underscore. But we shouldn't introduce more of them because those are actually reserved according to the C standard.